### PR TITLE
Add Logo option UwU

### PR DESCRIPTION
**Pull request by HEIGE-PCloud**
## Overview

Adds a query param to switch the Haskell logo to https://github.com/SAWARATSUKI/ServiceLogos:

- on: `?uwu=true`
- off: `?uwu=false`
Setting will persist until you turn it off.

## Screen

327063873-4bb93066-6606-4fab-9c5c-c54c94836630

## Credits 

Credit to @SAWARATSUKI in https://github.com/SAWARATSUKI/ServiceLogos

## Context

- https://react.dev/?uwu=true
- https://hono.dev/?kawaii=true
- https://twitter.com/saltyAom/status/1785339588661461365

### CODE REVIEW

File: site/templates/default.html
Comment (Kleidukos) [inline site/templates/default.html:21]: `localStorage.getItem` doesn't throw exception, if memory serves well. Why the try-catch block?

File: site/templates/default.html
Comment (Kleidukos) [inline site/templates/default.html:34]:
```suggestion
                  console.log('alternative logo enabled. turn off with ?uwu=false')
```

File: site/css/hl.css
Comment (Kleidukos) [inline site/css/hl.css:306]: please make the classes' names clearer, like `alternative-logo`

File: site/templates/default.html
Comment (Kleidukos) [inline site/templates/default.html:35]: Not blocking, but it might be better to have a separate page for acknowledgements on the website.

**Review by Kleidukos — CHANGES_REQUESTED** [review]

**Comment by tomjaguarpaw** [comment]
Could someone explain what this is about? I don't understand.

**Comment by Kleidukos** [comment]
@tomjaguarpaw Someone made cute alternative logos of projects: https://github.com/SAWARATSUKI/ServiceLogos

Some of them are already using them.

That being said since offering the logo from a third-party person is not a trivial matter I'd feel better if the committee would vote on this.

**Comment by HEIGE-PCloud** [comment]
@tomjaguarpaw It's basically a meme. Feel free to close it if this does not align well with the project.

**Comment by aviaviavi** [comment]
What's the license associated this logo?

Fun and lighthearted memes like this seem fine, very low risk, and a nice little social media opportunity - if the logo is permissive licensed. Dealing with any license obligations just for this seems unnecessary though.

**Comment by Tritlo** [comment]
I agree, I think it would be fun. The license is CC BY-NC-SA, so we should be good: https://github.com/SAWARATSUKI/ServiceLogos?tab=readme-ov-file#license. I suggest we move quickly on this, could be good PR as you say.

**Comment by Kleidukos** [comment]
I don't disagree with including the logo, I just want to focus on maintainability of the code. :)

**Comment by mpilgrem** [comment]
Given the general importance of logos/trade marks, I suggest that this is not merged until the Haskell.org Committee and/or the board of directors of Haskell.org, Inc. have had a chance to consider the implications of doing so.

**Comment by Tritlo** [comment]
The "collaborators" here are the committee 😄 

But I agree, we should have consensus. @rebeccaskinner, @aaronallen8455, @angerman, thoughts?

File: site/templates/default.html
Comment (Tritlo) [reply site/templates/default.html:21]: Indeed, it just returns null.

**Comment by mpilgrem** [comment]
One thing I would like the Haskell.org Committee to reflect upon is the use of the 'bag of money' iconography in the image. As far as I can see, none of the other examples has something similar.

Also, I do not read Japanese. Is there any perceived meaning to the Japanese characters included in the image?

**Comment by Tritlo** [comment]
> One thing I would like the Haskell.org Committee to reflect upon is the use of the 'bag of money' iconography in the image. As far as I can see, none of the other examples has something similar.

That's a good point. I'll tweak it and remove the moneybag.

**Comment by juhp** [comment]
> That's a good point. I'll tweak it and remove the moneybag.

It would make more sense to replace the bag with `<$>`...

(This comment is in no way an endorsement;)

File: site/templates/default.html
Comment (HEIGE-PCloud) [reply site/templates/default.html]: Not that it matters but I believe it is possible to disable localStorage under certain privacy settings, then it will throw something like `Uncaught DOMException: Failed to read the 'localStorage' property from 'Window': Access is denied for this document.` in Chrome. You can try it yourself in Chrome by changing the settings at Settings > Privacy and security > Site settings > Additional content settings > On-device site data > Don't allow sites to save data on your device

**Comment by Tritlo** [comment]
> > That's a good point. I'll tweak it and remove the moneybag.
> 
> It would make more sense to replace the bag with `<$>`...
> 
> (This comment is in no way an endorsement;)

Alright! I updated it again, moving the `<$>` in place of the money bag,

File: site/templates/default.html
Comment (Tritlo) [inline site/templates/default.html:31]:
```suggestion
                  console.log('logo credit to @sawaratsuki1004 via https://github.com/SAWARATSUKI/ServiceLogos. Slightly tweaked by @Tritlo');
```

File: site/css/hl.css
Comment (Tritlo) [inline site/css/hl.css:297]:
```suggestion
.alternative-logo-visible {
```

File: site/css/hl.css
Comment (Tritlo) [inline site/css/hl.css:300]:
```suggestion
.alternative-logo .alternative-logo-visible,.alternative-logo-hidden {
```

File: site/css/hl.css
Comment (Tritlo) [inline site/css/hl.css:305]:
```suggestion
.alternative-logo .alternative-logo-hidden {
```

**Review by Tritlo — COMMENTED** [review]
Change uwu to alternative-logo in the CSS.

File: site/templates/default.html
Comment (Tritlo) [reply site/templates/default.html]: Indeed, but that should be caught by the handler around the whole function, right?

**Comment by rhendric** [comment]
I don't want the OP to see my drive-by thumbs-down as any sort of personal attack, but I'd like the committee to consider whether they'd give equal consideration to an alternate Haskell logo that used iconography from a reality TV show, or that emulated a cryptocurrency marketing page, or that featured the [two soyjacks pointing](https://knowyourmeme.com/memes/two-soyjaks-pointing) meme. All of these, including the proposal, carry a certain amount of cultural baggage with them, and that baggage is not universally positive even if many people would consider them ‘fun’. Speaking strictly for myself, and without condemnation for anyone who feels differently, my emotional response to VTuber culture is negative, and so my emotional response to this logo being incorporated in a permanent way, even as an Easter egg, in the Haskell website is negative.

I'd be okay with this if it were a time-limited thing, either as part of a rotation like Google Doodles, or as an April 1 stunt (or to commemorate a day or event that is special for VTuber culture). To have it be ‘the’ alternate official haskell.org logo in all perpetuity just gives me the wibbles.

File: site/templates/default.html
Comment (HEIGE-PCloud) [reply site/templates/default.html]: > Indeed, but that should be caught by the handler around the whole function, right?

The idea is that we still want to set the class name regardless whether we have successfully read the localStorage value. i.e. if `?qwq=true` is set in the url, then the logo is guaranteed to be switched, no matter if the localStorage is working or not.

**Comment by Abastro** [comment]
Huh, why is this 'uwu'? Sorry that I am missing a lot.

**Comment by Kleidukos** [comment]
The repo and the original images (and licenses) have disappeared. I'm afraid this is going to be blocked until we can reliably get a source and license for the images. Also I wonder what motivated the deletion of the repo.

**Comment by Tritlo** [comment]
The author had this to say on Twitter:

 > To clean up the license, make the repository private and create a new one.
> 
> Please wait until it is re-released 🫠

image

So I guess we'll wait until the license is clarified.

**Comment by kutyel** [comment]
> The repo and the original images (and licenses) have disappeared. I'm afraid this is going to be blocked until we can reliably get a source and license for the images. Also I wonder what motivated the deletion of the repo.

The logos will come back soon with I presume a more open license, I think the repo will be this: https://github.com/SAWARATSUKI/Logos (empty for now)

**Comment by SAWARATSUKI** [comment]
I am the creator of the logo
I created this as pure fan art with no intention of dissatisfaction or opposition to the Haskell people.
First of all, I'll admit if I made you dissatisfied.

If the official uses it, you are free to use it for anything, including commercial use!🫠
I plan to make it impossible to use it for commercial purposes other than official use!

**Comment by angerman** [comment]
I'm not opposed to this. I did see the one with the money bag before and found that a bit questionable too. But the current one, I have no objections to. We all need a bit of fun in life. ✌️

**Comment by SAWARATSUKI** [comment]
I only used 💰 because I remembered hearing from a friend that Haskell is used in banking and added it.
My apologies.
I'm glad too that you like the modified logo 🫠.
Please use it as you wish!

**Comment by HEIGE-PCloud** [comment]
I am considering to start using

```hs
(💰) = ($)
```

from now on 👿

**Comment by kutyel** [comment]
@Kleidukos if we have written permission from the author of the logos to use the modified version is there a reason to keep holding this? 👀

**Comment by Tritlo** [comment]
We still need a green light from @rebeccaskinner and @tomjaguarpaw. Are you ok with this?

**Comment by tomjaguarpaw** [comment]
I'm going to abstain. I still don't really understand what this is.

**Comment by hmemcpy** [comment]
Haskell: Avoid fun at all cost

just merge this already! :D

**Comment by ekmett** [comment]
I just wanted to chime in and say that this logo is all kinds of awesome. Great work.

**Comment by ekmett** [comment]
> I am considering to start using
> 
> ```haskell
> (💰) :: (a -> b) -> a -> b
> (💰) = ($)


If you're going to make another operator, you might try flipping the associativity, along with an appropriate (💰!) for strict application.

```
with that 💰 you can use 💰 it to divide 💰! multi-part arguments = 
with that (you can use) (it to divide) $! multi-part arguments
```

and say which ones you want to apply strictly, and . still works, so now the . vs $ debate goes away, as there is now a canonical choice. That said, having built a language that used it that way, I er.. eventually relented and switched back because the muscle memory was too strong for the other meaning.

**Review by rebeccaskinner — APPROVED** [review]
I'm not familiar with the meme, but I'm happy for haskell.org to participate in a fun thing and this seems harmless.

Tested this with javascript enabled and disabled on Firefox and Chromium, didn't see any adverse impact and it looks straightforward.

**Comment by aaronallen8455** [comment]
Fwiw, I found that adding `filter: drop-shadow(0px 0px 1px black)` to the CSS rules for the image makes it look a bit better against a white background:
Screenshot 2024-05-05 at 2 19 29 AM

**Comment by Tritlo** [comment]
Seems like we have consensus! @Kleidukos, are you satisfied with the licensing after the written permission?

I'll integrate the drop-shadow @aaronallen8455, and do one last test on whether we need the try-catch for localstorage for some permission configurations.

**Comment by Tritlo** [comment]
> Seems like we have consensus! @Kleidukos, are you satisfied with the licensing after the written permission?

(I need you to approve the changes in the GitHub interface also)

**Review by Kleidukos — APPROVED** [review]
LGTM.

**Comment by Tritlo** [comment]
It's up! Thanks everyone.

**Comment by Kleidukos** [comment]
Official announcement: https://discourse.haskell.org/t/cute-alternative-logo-for-haskell-org/9473

**Comment by MRandl** [comment]
It renders in a weird manner on my phone. Not a big deal but I thought you should know

**Comment by Tritlo** [comment]
> It renders in a weird manner on my phone. Not a big deal but I thought you should know
> 
> 
> 
> 
> 
> Pixel 7a on firefox

It's been fixed, just waiting to merge: https://github.com/haskell-infra/www.haskell.org/pull/294

**Comment by MRandl** [comment]
Oh I didn't know. Sorry to have bothered !

**Comment by Tritlo** [comment]
> Oh I didn't know. Sorry to have bothered!

No, it's good. I thought it was only on iOS, seems like it is for all mobile.

**Pull request merged**
